sfyra/pkg/tests/server_class.go

```diff
@@ -41,6 +42,23 @@ func TestServerClassDefault(ctx context.Context, metalClient client.Client, vmSe
 		serverClass, err := createServerClass(ctx, metalClient, serverClassName, classSpec)
 		require.NoError(t, err)
 
+		// verify sc is initially empty
+		assert.Len(t, append(serverClass.Status.ServersAvailable, serverClass.Status.ServersInUse...), 0)
```

Comment:
:+1:

Reply:
Only thing I might suggest is that we do something similar to this but in the server test. It will kick off the server reset sooner. I don't think it matters too much in the end.

Reply:
this condition will fail if we re-run the tests (server class will already grab some servers)

my only suggestion is to create and destroy set of "fake" servers specifically to test this feature

Reply:
That makes sense.